lib/husky_musher/app.py

```diff
@@ -127,21 +135,34 @@ def main():
         # If not in REDCap project, create new record
         user_data = {'record_id': register_net_id(net_id)}
 
-    # TODO -- generate a survey link for a particular day
-    # We are awaiting finalization of the REDCap project to know how
-    # daily attestations (repeating instruments) will be implemented.
+    # Because of REDCap's survey queue logic, we can point a participant to an
+    # upstream survey. If they've completed it, REDCap will automatically direct
+    # them to the next, uncompleted survey in the queue.
+    event = 'enrollment_arm_1'
+    instrument = 'eligibility_screening'
+    repeat_instance = None
+
+    # If all enrollment event instruments are complete, point participants
+    # to today's daily attestation instrument.
+    # If the participant has already completed the daily attestation,
+    # REDCap will prevent the participant from filling out the survey again.
     if is_complete('eligibility_screening', user_data) and \
         is_complete('consent_form', user_data) and \
         is_complete('enrollment_questionnaire', user_data):
-        return f"Congrats, {net_id}, you're already registered under record ID " \
-            f"{user_data['record_id']} and your eligibility " \
-            "screening, consent form, and enrollment questionnaires are complete!"
 
-    # Generate a link to the eligibility questionnaire, and then redirect.
-    # Because of REDCap's survey queue logic, we can point a participant to an
-    # upstream survey. If they've completed it, REDCap will automatically direct
-    # them to the next, uncompleted survey in the queue.
-    return redirect(generate_survey_link(user_data['record_id']))
+        event = 'encounter_arm_1'
+        instrument = 'daily_attestation'
+        # Repeat instance number should be days since the start of the study,
+        # with the first instance starting at 1.
+        repeat_instance = 1 + (datetime.today() - STUDY_START_DATE).days
+
+        if repeat_instance <= 0:
+            # This should never happen!
+            app.logger.warning("Failed to create a valid repeat instance")
```

Comment:
Can we raise an `app.logger.error` here?

Reply:
Ah, yes!